provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -469,6 +466,13 @@ func (c *Client) addNetworkDevice(
 	wakeOnLan := true
 	networkDevice.WakeOnLanEnabled = &wakeOnLan
 	networkDevice.Backing = networkBacking
+	if mac != "" {
+		if !VerifyMAC(mac) {
+			return nil, fmt.Errorf("Invalid MAC address: %q", mac)
```

Comment:
Let's return a short snippet of what a valid mac looks like. Also consider adding snippets to verifymac and return specific errors on where it failed validation.